components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/common/SynapsePropertiesHandler.java

```diff
@@ -61,6 +77,62 @@ public boolean handleRequest(MessageContext messageContext) {
     }
 
     public boolean handleResponse(MessageContext messageContext) {
+        if (config == null) {
+            // Retrieve the ISKMReverseProxyEnabled property value from api manager configurations.
+            // This value indicates whether the IS Authentication endpoint has been reverse proxied through
+            // the Gateway.
+            config = ServiceReferenceHolder.getInstance().getApiManagerConfigurationService()
+                    .getAPIManagerConfiguration();
+            iskmReverseProxyEnabled = Boolean.parseBoolean(
+                    config.getFirstProperty(APIConstants.AUTH_MANAGER + APIConstants.IS_KM_REVERSE_PROXY_ENABLED));
+        }
+        // Modify location header only if ISKMReverseProxyEnabled property is set to true
+        if (iskmReverseProxyEnabled) {
+            // The logic is related if the API context is "/authorize", "/commonauth" or "/oidc" while status code is 302
+            if (messageContext.getProperty(RESTConstants.REST_API_CONTEXT)
```

Comment:
Shall we change it as constant.equals(variable) instead of variable.equals(constant)?

Reply:
Fixed by https://github.com/wso2/carbon-apimgt/pull/9734/commits/6861d44172fedbddb4c2f97439bf41943a2ff0bc